src/pages/Environments.js

```diff
@@ -425,9 +425,13 @@ const Environments = () => {
               const multipleDisksOfType = _.remove(disk => getDiskAppType(disk) !== appType || disk.status === 'Deleting',
                 disksByProject[googleProject]).length > 1
               const forAppText = !!appType ? ` for ${_.capitalize(appType)}` : ''
+              // Galaxy can have multiple disks per project as long as they are in different workspaces
+              const galaxyDisksOnly = _.filter(disk => getDiskAppType(disk) === tools.galaxy.appType && disk.status !== 'DELETING', filteredDisks)
+              const galaxyDiskWorkspaces = _.map(currentDisk => currentDisk.labels.saturnWorkspaceName, galaxyDisksOnly)
+              const multipleGalaxyDisks = _.uniq(galaxyDiskWorkspaces).length !== galaxyDiskWorkspaces.length
               return h(Fragment, [
                 googleProject,
-                diskStatus !== 'Deleting' && multipleDisksOfType &&
+                diskStatus !== 'Deleting' && (appType === tools.galaxy.appType ? multipleGalaxyDisks : multipleDisksOfType) &&
                 h(TooltipTrigger, {
                   content: `This billing project has multiple active persistent disks${forAppText}. Only the latest one will be used.`
                 }, [icon('warning-standard', { style: { marginLeft: '0.25rem', color: colors.warning() } })])
```

Comment:
Within the `cellRenderer` function, while evaluating whether we have multiple disks, we seem to be running both of the evaluation logic (one for Galaxy apps and one for non-Galaxy apps). This should be unnecessary since an app cannot be Galaxy and non-Galaxy at the same time. In other words we could move the conditional from the rendering code to higher up and avoid the unnecessary computations, like so:

```javascript
            cellRenderer: ({ rowIndex }) => {
              const { status: diskStatus, googleProject } = filteredDisks[rowIndex]
              const appType = getDiskAppType(filteredDisks[rowIndex])
              const haveMultipleDisks = () => {
                if (appType === tools.galaxy.appType) {
                  // Galaxy can have multiple disks per project as long as they are in different workspaces
                  const galaxyDisksOnly = _.filter(disk => appType === tools.galaxy.appType && disk.status !== 'DELETING', filteredDisks)
                  const galaxyDiskWorkspaces = _.map(currentDisk => currentDisk.labels.saturnWorkspaceName, galaxyDisksOnly)
                  return _.uniq(galaxyDiskWorkspaces).length !== galaxyDiskWorkspaces.length
                } else {
                  return _.remove(disk => getDiskAppType(disk) !== appType || disk.status === 'Deleting', disksByProject[googleProject]).length > 1
                }
              }

              const forAppText = !!appType ? ` for ${_.capitalize(appType)}` : ''

              return h(Fragment, [
                googleProject,
                diskStatus !== 'Deleting' && haveMultipleDisks() &&
                h(TooltipTrigger, {
                  content: `This billing project has multiple active persistent disks${forAppText}. Only the latest one will be used.`
                }, [icon('warning-standard', { style: { marginLeft: '0.25rem', color: colors.warning() } })])
              ])
            }
```

What do you think?

Reply:
I have made the PR #2750 into your branch to better illustrate the change above and also pull the logic into its own function, which would also make it possible to write a unit test for it.